ftw/solr/connection.py

```diff
@@ -161,13 +163,31 @@ def filter_extract_commands(self):
             added.add((uid, field))
         self.extract_commands = reversed(filtered_extract_commands)
 
+    def updates_in_post_commit_enabled(self):
+        registry = queryUtility(IRegistry)
+        settings = registry.forInterface(ISolrSettings)
+        return settings.enable_updates_in_post_commit_hook
+
     def flush(self, extract_after_commit=True):
         """Send queued update commands to Solr."""
         if self.update_commands:
             data = '{%s}' % ','.join(self.update_commands)
-            resp = self.post('/update', data=data, log_error=False)
-            if not resp.is_ok():
-                logger.error('Update command failed. %s', resp.error_msg())
+
+            if self.updates_in_post_commit_enabled():
+                def hook(succeeded, data):
+                    if not succeeded:
+                        return
+                    resp = self.post('/update', data=data, log_error=False)
+                    if not resp.is_ok():
+                        logger.error('Post commit update command failed. %s',
+                                     resp.error_msg())
+                transaction.get().addAfterCommitHook(
```

Comment:
This brakes reindexing using the maintenance view. The maintenance view does not commit the ZODB transaction to avoid any read conflict errors. To handle this case there's an `extract_after_commit` flag which is set to `False` for the extract commands that are processed in a post commit hook as well: https://github.com/4teamwork/ftw.solr/blob/master/ftw/solr/connection.py#L219

I would rename the flag to `after_commit` and use it for both commands.